src/python/grpcio_tests/tests/interop/methods.py

```diff
@@ -291,36 +315,89 @@ def _empty_stream(stub):
+            messages_pb2.ResponseParameters(size=1),),
+        payload=messages_pb2.Payload(body=b'\x00'),
+        response_status=messages_pb2.EchoStatus(code=_CODE, message=_MESSAGE))
+    pipe.add(request)   # sends the initial request.
+  # Dropping out of with block closes the pipe
+  ValidateStatusCodeAndMessage(response_iterator)
+
+
+def _unimplemented_method(stub):
+  status = grpc.StatusCode.UNIMPLEMENTED
+  response_future = stub.UnimplementedCall.future(empty_pb2.Empty())
   if response_future.code() != status:
     raise ValueError(
       'expected code %s, got %s' % (status, response_future.code()))
-  elif response_future.details() != message:
-    raise ValueError(
-      'expected message %s, got %s' % (message, response_future.details()))
 
-  request = messages_pb2.StreamingOutputCallRequest(
+def _custom_metadata(stub):
+  # Function wide constants
+  _INITIAL_METADATA_VALUE = "test_initial_metadata_value"
+  _TRAILING_METADATA_VALUE = "\x0a\x0b\x0a\x0b\x0a\x0b"
+  _METADATA = [
+      (_INITIAL_METADATA_KEY, _INITIAL_METADATA_VALUE),
+      (_TRAILING_METADATA_KEY, _TRAILING_METADATA_VALUE)]
+
+  def ValidateMetadata(response):
+    initial_metadata = dict(response.initial_metadata())
+    if initial_metadata[_INITIAL_METADATA_KEY] != _INITIAL_METADATA_VALUE:
+      raise ValueError(
+        'expected initial metadata %s, got %s' % (
+            _INITIAL_METADATA_VALUE, initial_metadata[_INITIAL_METADATA_KEY]))
+    trailing_metadata = dict(response.trailing_metadata())
+    if trailing_metadata[_TRAILING_METADATA_KEY] != _TRAILING_METADATA_VALUE:
+      raise ValueError(
+        'expected trailing metadata %s, got %s' % (
+            _TRAILING_METADATA_VALUE, initial_metadata[_TRAILING_METADATA_KEY]))
+
+  # Testing with UnaryCall
+  request = messages_pb2.SimpleRequest(
       response_type=messages_pb2.COMPRESSABLE,
-      response_parameters=(
-          messages_pb2.ResponseParameters(size=1),),
-      response_status=messages_pb2.EchoStatus(code=code, message=message))
-  response_iterator = stub.StreamingOutputCall(request)
-  if response_future.code() != status:
-    raise ValueError(
-      'expected code %s, got %s' % (status, response_iterator.code()))
-  elif response_future.details() != message:
-    raise ValueError(
-      'expected message %s, got %s' % (message, response_iterator.details()))
+      response_size=1,
+      payload=messages_pb2.Payload(body=b'\x00'))
+  response_future = stub.UnaryCall.future(request, metadata=_METADATA)
+  ValidateMetadata(response_future)
 
+  # Testing with FullDuplexCall
+  with _Pipe() as pipe:
+    response_iterator = stub.FullDuplexCall(pipe, metadata=_METADATA)
+    request = messages_pb2.StreamingOutputCallRequest(
+        response_type=messages_pb2.COMPRESSABLE,
+        response_parameters=(
+            messages_pb2.ResponseParameters(size=1),))
+    pipe.add(request)   # Sends the request
+    next(response_iterator)    # Causes server to send trailing
```

Comment:
"trailing" isn't a noun, as we use the term throughout the codebase - do you mean "trailing metadata"?